src/System.Private.Xml.Linq/src/System/Xml/Linq/XHashtable.cs

```diff
@@ -109,17 +109,8 @@ public TValue Add(TValue value)
                 lock (this)
                 {
                     XHashtableState newState = _state.Resize();
-
                     // Use memory barrier to ensure that the resized XHashtableState object is fully constructed before it is assigned
-#if !SILVERLIGHT 
                     Thread.MemoryBarrier();
```

Comment:
Looks like you need a `using System.Threading;` at the top of the file.  You may also need `<Reference Include="System.Threading.Thread" />` in the .csproj.

Reply:
addressed